proto/osmosis/protorev/v1beta1/protorev.proto

```diff
@@ -73,23 +73,62 @@ message RouteStatistics {
   repeated uint64 route = 3 [ (gogoproto.moretags) = "yaml:\"route\"" ];
 }
 
-// PoolWeights contains the weights of all of the different pool types. This
-// distinction is made and necessary because the execution time ranges
-// significantly between the different pool types. Each weight roughly
-// corresponds to the amount of time (in ms) it takes to execute a swap on that
-// pool type.
-message PoolWeights {
+// InfoByPoolType contains information pertaining to how expensive (in terms of
+// gas and time) it is to execute a swap on a given pool type. This distinction
+// is made and necessary because the execution time ranges significantly between
+// the different pool types.
+message InfoByPoolType {
+  // The stable pool info
+  StablePoolInfo stable = 1 [
+    (gogoproto.moretags) = "yaml:\"stable\"",
+    (gogoproto.nullable) = false
+  ];
+  // The balancer pool info
+  BalancerPoolInfo balancer = 2 [
+    (gogoproto.moretags) = "yaml:\"balancer\"",
+    (gogoproto.nullable) = false
+  ];
+  // The concentrated pool info
+  ConcentratedPoolInfo concentrated = 3 [
+    (gogoproto.moretags) = "yaml:\"concentrated\"",
+    (gogoproto.nullable) = false
+  ];
+  // The cosmwasm pool info
+  CosmwasmPoolInfo cosmwasm = 4 [
+    (gogoproto.moretags) = "yaml:\"cosmwasm\"",
+    (gogoproto.nullable) = false
+  ];
+}
+
+// StablePoolInfo contains meta data pertaining to a stableswap pool type.
+message StablePoolInfo {
   // The weight of a stableswap pool
-  uint64 stable_weight = 1 [ (gogoproto.moretags) = "yaml:\"stable_weight\"" ];
+  uint64 weight = 1 [ (gogoproto.moretags) = "yaml:\"weight\"" ];
+}
+
+// BalancerPoolInfo contains meta data pertaining to a balancer pool type.
+message BalancerPoolInfo {
   // The weight of a balancer pool
-  uint64 balancer_weight = 2
-      [ (gogoproto.moretags) = "yaml:\"balancer_weight\"" ];
+  uint64 weight = 1 [ (gogoproto.moretags) = "yaml:\"weight\"" ];
+}
+
+// ConcentratedPoolInfo contains meta data pertaining to a concentrated pool
+// type.
+message ConcentratedPoolInfo {
   // The weight of a concentrated pool
-  uint64 concentrated_weight = 3
-      [ (gogoproto.moretags) = "yaml:\"concentrated_weight\"" ];
-  // The weight of a cosmwasm pool
-  uint64 cosmwasm_weight = 4
-      [ (gogoproto.moretags) = "yaml:\"cosmwasm_weight\"" ];
+  uint64 weight = 1 [ (gogoproto.moretags) = "yaml:\"weight\"" ];
+  // The maximum number of ticks we can move when rebalancing
+  uint64 max_ticks_crossed = 2
+      [ (gogoproto.moretags) = "yaml:\"max_ticks_crossed\"" ];
+}
+
+// CosmwasmPoolInfo contains meta data pertaining to a cosmwasm pool type.
+message CosmwasmPoolInfo {
+  // The weight of a cosmwasm pool (by contract address)
+  map<string, uint64> weight_map = 1 [
```

Comment:
I'm not sure if using a map is problematic for app hashing.

Reply:
@ValarDragon can you verify the safety of this?

Reply:
We had instances where we tried to use maps quite a while ago, I think the decision back then was to try avoid using maps for extra safety, would it be possible to change it so tht we use repeated structs instead here?

Reply:
got it. ty for confirming